File: spec/construct/test_optional_id.py
import unittest

from optional_id import _schema

class TestOptionalId(unittest.TestCase):
    def test_optional_id(self):
        r = _schema.parse_file('src/fixed_struct.bin')

        self.assertEqual(r._unnamed0, 80)
        self.assertEqual(r._unnamed1, 65)
        self.assertEqual(r._unnamed2, b"\x43\x4B\x2D\x31\xFF")

File: spec/rust/tests/test_optional_id.rs
use std::fs;
extern crate kaitai;
use self::kaitai::*;
use rust::formats::optional_id::*;

#[test]
fn test_optional_id() -> KResult<()> {
    let bytes = fs::read("../../src/fixed_struct.bin").unwrap();
    let _io = BytesReader::from(bytes);
    let r: OptRc<OptionalId> = OptionalId::read_into(&_io, None, None)?;

    assert_eq!(*r.unnamed0(), 80);
    assert_eq!(*r.unnamed1(), 65);
    assert_eq!(*r.unnamed2(), vec![0x43, 0x4B, 0x2D, 0x31, 0xFF]);
    Ok(())
}
